_posts/2019-04-10-aws-dashboard-console-using-arduino.md

```diff
@@ -0,0 +1,265 @@
+
+### Circuit of the Console
+
+The circuit of the console contains following components:
+- 1 x NodeMCU board
+- 1 x LCD 2x16 characters I2C
+- 7 x LED lights
+- 7 x resistors for the LEDs
+- 1 x breadboard
+- wires
+
+![Dashboard circuit](/img/aws-dashboard-console-using-arduino/dashboard-circuit.png "Dashboard circuit")
+
+Basic layout of the circuit is a relative simple. The LEDs (and resistors) are connected to the NodeMCU IO pins. The LCD screen data wires are connected to the default I2C IO pins. The LCD screen power is connected to the NodeMCU 5V VIN pin. The NodeMCU board is powered by the USB connector. Components are assembled on the breadboard to minimize soldering.
+
+> *Note that this circuit contains a conflict: the NodeMCU board runs by an 3V internal voltage and IO pins are therefore 3V tolerant. On opposite, the LCD screen requires a 5V voltage. Related to the documentation, this circuit should not work since the 5V LCD data wires are connected to the 3V NodeMCU pins. Luckily I found blog posts where this kind of the circuit was proven to work. At least my version of the NodeMCU seems to work fine event the 3V tolerant IO pins are driven by the 5V voltage.*
+
+### The Arduino program
+
+The Arduino framework programming model contains two main methods: `setup` and `loop`. The `setup` method is run once when the board is powered. The `loop` method is run infinitely. In addition, you can define own custom methods.
+
+Let’s look at some parts of the code:
+
+```c++
+#define ALARMS_LED D0
+#define PIPE_RUNNING_LED_DEV D8
+#define PIPE_FAILED_LED_DEV D7
+#define PIPE_RUNNING_LED_TEST D6
+#define PIPE_FAILED_LED_TEST D5
+#define PIPE_RUNNING_LED_PROD D4
+#define PIPE_FAILED_LED_PROD D3
+
+void setup() {
+  Serial.begin(115200);
+  
+  pinMode(ALARMS_LED, OUTPUT);
+  pinMode(PIPE_RUNNING_LED_DEV, OUTPUT);
+  pinMode(PIPE_FAILED_LED_DEV, OUTPUT);
+  pinMode(PIPE_RUNNING_LED_TEST, OUTPUT);
+  pinMode(PIPE_FAILED_LED_TEST, OUTPUT);
+  pinMode(PIPE_RUNNING_LED_PROD, OUTPUT);
+  pinMode(PIPE_FAILED_LED_PROD, OUTPUT);
+
+  Serial.print("START");
+
+  lcd.init();
+  lcd.backlight();
+
+  initConsole();
+  connectWifi();
+  
+  delay(2000);
+  lcd.setCursor(0, 0);
+  lcd.print("Call AWS...     ");
+  lcd.setCursor(0, 1);
+  lcd.print("                ");
+}
+```
+
+First, the LED pins are defined. This definition is required to setup which IO pins the LED wires are connected. Then the IO pin mode is set to “OUTPUT” which is required to enable a write mode for the pins.
+
+The LCD screen is initialized and backlight is switch on. The electronic components are initialized by the custom method `initConsole`. Basically the `initConsole` will blink the LEDs and write some text to the screen. Finally, the WiFi is connected.
```

Comment:
and backlight is switch on --> and the backlight is switched on?

Reply:
Basically the `initConsole` will blink -->
Basically, the `initConsole` will blink
Tai pelkästään 
The `initConsole` will blink